lib/lumberjack/beats/server.rb

```diff
@@ -66,8 +66,16 @@ def run(&block)
         # unless the server is closing
         next unless connection
 
+        Thread.abort_on_exception = true
         Thread.new(connection) do |connection|
-          connection.run(&block)
+          begin
+            connection.run(&block)
+          rescue Lumberjack::Beats::Connection::ConnectionClosed 
+            # Connection will raise a wrapped exception upstream, 
+            # but if the threads are managed by the library we can simply ignore it.
```

Comment:
I'm not very familiar with the insides so the "if the threads are managed...we can simply ignore it" is strange, what if they're not? also, by library you mean ruby-lumberjack?